### PR TITLE
feat: hint transformed id on model and pipeline creation

### PR DESCRIPTION
**Pull request by EiffelFly**
Because

- hint transformed id on model and pipeline creation
CleanShot 2024-09-02 at 11 05 10 2@2x

This commit

- hint transformed id on model and pipeline creation

### CODE REVIEW

File: packages/toolkit/src/view/recipe-editor/popovers/PipelineNamePopover.tsx [hunk 1]
```diff
@@ -72,13 +77,16 @@ export const PipelineNamePopover = ({
     });
   }, [routeInfo.isSuccess, routeInfo.data.resourceId, form, pipelineIsPublic]);
 
+  const formattedPipelineId = formatResourceId(form.watch("id"), "p");
```
Comment (orangecoloured) [inline packages/toolkit/src/view/recipe-editor/popovers/PipelineNamePopover.tsx:80]: Should we move `p`, `m`, `c` etc. to a separate constants file?

**Review by orangecoloured — CHANGES_REQUESTED** [review]

File: packages/toolkit/src/view/recipe-editor/popovers/PipelineNamePopover.tsx (hunk 1 above)
Comment (EiffelFly) [reply packages/toolkit/src/view/recipe-editor/popovers/PipelineNamePopover.tsx:80]: Oh that is a good suggestion, I will do that 🤝

**Review by thewbuk — DISMISSED** [review]
LGTM

**Comment by orangecoloured** [comment]
@EiffelFly also, how about we do `p_bla-bla`? With a `_` separating the prefix? I mean if t's a prefix, makes sense to have it easily visible.

**Comment by EiffelFly** [comment]
@orangecoloured 

> also, how about we do p_bla-bla? With a _ separating the prefix? I mean if t's a prefix, makes sense to have it easily visible.

Oh yes, that is a good suggestion too. But because we don't allow underscore in some place, I think we can make it like

`p-`, `c-` and `m-`

**Pull request merged**